openupgrade_scripts/scripts/sale_purchase/16.0.1.0/post-migration.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2023 Coop IT Easy SC
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+
+def convert_service_to_purchase_to_company_dependent(env):
+    # the product.template.service_to_purchase field became company-dependent.
+    # this means that it is no longer stored in the product_template table,
+    # but in the ir_property table.
+    #
+    # for such cases, openupgrade.convert_to_company_dependent() should
+    # normally be used, but that function does not seem to support converting
+    # a field to company-dependent without changing its name at the same time.
+    # moreover, it stores boolean values even when they are false (what odoo
+    # does not), and it creates values for all companies, which does not make
+    # sense when a record is linked to a particular company only.
```

Comment:
This is something to be put as docstring of the method IMO, as it describes what the method is going to do.

```suggestion
    """The product.template.service_to_purchase field became company-dependent.
    this means that it is no longer stored in the product_template table,
    but in the ir_property table.
    
    For such cases, openupgrade.convert_to_company_dependent() should
    normally be used, but that function does not seem to support converting
    a field to company-dependent without changing its name at the same time.
    moreover, it stores boolean values even when they are false (what odoo
    does not), and it creates values for all companies, which does not make
    sense when a record is linked to a particular company only.
    """
```

Reply:
Anyway, your considerations are not totally correct:

> but that function does not seem to support converting a field to company-dependent without changing its name at the same time

I don't fully understand what you mean, but the usual method is to rename the origin column in pre-migration (for preservation purposes, but also for not confusing people browsing DB schema thinking that the applied value is the one in the column), and then call this other method in post.

> it stores boolean values even when they are false

This can be considered a bug of the method (or something that has changed across versions, and should be split the behavior inside the method), but it should be fixed there in openupgradelib.

> It creates values for all companies, which does not make sense when a record is linked to a particular company only.

This is a temporary condition. If you empty the company field after migration, you expect to have the same value for the rest of the companies as it was previous the migration, so we can't limit it.

Maybe for not over-heading the table, a ir.property without company_id can be created for this, but we should not how Odoo acts when you change the value in one company in this case. Is it modified for all the companies as the existing `ir.property` has no company?

So in summary, we shouldn't add so many code for something that is a recurrent problem and should be handled in the library.